modules/tensor_mechanics/src/materials/HillConstants.C

```diff
@@ -239,4 +327,23 @@ HillConstants::rotateHillConstants(const std::vector<Real> & hill_constants_inpu
                        L * _transformation_tensor(3, 4) * _transformation_tensor(3, 4) +
                        M * _transformation_tensor(3, 5) * _transformation_tensor(3, 5) +
                        N * _transformation_tensor(3, 3) * _transformation_tensor(3, 3);
+
+  // Alternative approach for hill tensor rotation
+  // Both ways of computing anisotropic coefficient updates are carried out here
+  // and one is selected when computing the stress updates (for creep, e.g.)
+  _hill_tensor.zero();
+
+  _hill_tensor(0, 0) = G + H;
+  _hill_tensor(1, 1) = F + H;
+  _hill_tensor(2, 2) = F + G;
+  _hill_tensor(0, 1) = _hill_tensor(1, 0) = -H;
+  _hill_tensor(0, 2) = _hill_tensor(2, 0) = -G;
+  _hill_tensor(1, 2) = _hill_tensor(2, 1) = -F;
+
+  _hill_tensor(3, 3) = 2.0 * N;
+  _hill_tensor(4, 4) = 2.0 * L;
+  _hill_tensor(5, 5) = 2.0 * M;
+
+  _hill_tensor.right_multiply_transpose(_transformation_tensor);
+  _hill_tensor.left_multiply(_transformation_tensor);
```

Comment:
We should either rotate the `HillConstants` or the `HillTensor` depending on the rotation angle. We should add a selection criterion here.

Reply:
The `stressUpdate` material would need to know what's available, constants or tensor. If the material provide both the selection can take place in the constitutive model class. If we choose to rotate only one of them here, we may need to add another user input that would need to be coordinated with, e.g., the anisotropic creep class.

@dschwen any idea to make this design cleaner?

Reply:
How about we keep both material properties but only rotate whichever is necessary? We can build the updated Hill tensor from the rotated Hill constants for the first case.

Reply:
I think it can be involved to decide which transformation is necessary without adding flags to coordinate with the stress update classes (creep can work with constants, but plasticity doesn't really benefit much from constants since it performs tensor operations).  We know we cannot describe large deformation through Hill constants since they cannot be extracted from a generally transformed Hill's tensor. I added an if statement to avoid performing the extraction of these components when we know it won't be needed. This can save us some time when we choose to include large deformation rotation in the anisotropy directions.